libs/header/sync/sync.go

```diff
@@ -41,8 +42,8 @@ type Syncer[H header.Header] struct {
 	state   State
 	// signals to start syncing
 	triggerSync chan struct{}
-	// syncedHead is the latest synced header.
-	syncedHead H
+	// syncedHeadPtr is the latest synced header.
+	syncedHeadPtr atomic.Pointer[H]
```

Comment:
optional nit, but I think we should keep the name - the type information that it's a pointer is already in the type

Reply:
Ok. Will change it